sync/src/chain.rs

```diff
@@ -452,7 +453,11 @@ impl ChainSync {
 
 		// Disable the peer for this syncing round if it gives invalid chain
 		if !valid_response {
-			trace!(target: "sync", "{} Disabled for invalid headers response", peer_id);
+			trace!(target: "sync", "{} Deactivated for invalid headers response", peer_id);
+			self.deactivate_peer(io, peer_id); 
+		}
+		if headers.is_empty() {
```

Comment:
doesn't signal a misbehaving or useless peer - could be caused by other peers giving us blocks we didn't ask for.